File: Math/TypeScript/_1281_SubtracttheProductandSumofDigitsofanInteger.ts
function subtractProductAndSum(n: number): number {
  const nStr = n.toString();
  let pod = 1;
  let sod = 0;
  for (const ns of nStr) {
    pod *= Number(ns);
    sod += Number(ns);
  }
  return pod - sod;
}

console.log(subtractProductAndSum(234));
console.log(subtractProductAndSum(4421));

File: Math/Python/_1281_SubtracttheProductandSumofDigitsofanInteger.py
from functools import reduce


class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        pod = reduce((lambda x, y: x * y), map(int, str(n)))
        sod = reduce((lambda x, y: x + y), map(int, str(n)))
        return pod - sod


ins = Solution()

print(ins.subtractProductAndSum(234))
print(ins.subtractProductAndSum(4421))
